scripts/baseDemo.ts

```diff
@@ -0,0 +1,363 @@
+        id
+        title
+      }
+    }
+  }
+`);
+const CREATE_GRAPH = gql(`
+  mutation CreateDemoGraph ($accountId: ID!, $newServiceId: ID!, $name: String) {
+    newService(accountId: $accountId, id: $newServiceId, name: $name) {
+      id
+      apiKeys { token }
+      account {
+        currentPlan {
+          kind
+        }
+      }
+    }
+  }
+`);
+const CREATE_SUBGRAPH = gql(`
+  mutation PublishSubgraphSchema($graphId: ID!, $graphVariant: String!, $name: String!, $activePartialSchema: PartialSchemaInput!, $url: String, $revision: String!) {
+    graph(id: $graphId) {
+      publishSubgraph(graphVariant: $graphVariant, activePartialSchema: $activePartialSchema, name: $name, url: $url, revision: $revision) { 
+        launchUrl
+        updatedGateway
+        wasCreated
+      }
+    }
+  }
+`);
+const CREATE_OPERATION_COLLECTION = gql(`
+  mutation CreateOperationCollection($isSandbox: Boolean!, $isShared: Boolean!, $name: String!, $variantRefs: [ID!]) {
+    createOperationCollection(isSandbox: $isSandbox, isShared: $isShared, name: $name, variantRefs: $variantRefs) {
+      ... on OperationCollection {
+        id
+      }
+      ... on PermissionError {
+        message
+      }
+      ... on ValidationError {
+        message
+      }
+    }
+  }
+`);
+const ADD_OPERATION_TO_COLLECTION = gql(`
+  mutation AddOperations($operations: [AddOperationInput!]!, $operationCollectionId: ID!) {
+    operationCollection(id: $operationCollectionId) {
+      addOperations(operations: $operations) {
+        ... on ValidationError {
+          message
+        }
+        ... on PermissionError {
+          message
+        }
+        ... on AddOperationCollectionEntriesSuccess {
+          operationCollectionEntries {
+            id
+            name
+          }
+        }
+      }
+    }
+  }
+`);
+const UPDATE_EXPLORER_URL = gql(`
+  mutation UpdateURL($name: String!, $graphId: ID!, $url: String, $subscriptionUrl: String, $sharedHeaders: String) {
+    graph(id: $graphId) {
+      variant(name: $name) {
+        updateURL(url: $url) {
+          url
+        }
+        updateSubscriptionURL(subscriptionUrl: $subscriptionUrl) {
+          url
+        }
+        updateSharedHeaders(sharedHeaders: $sharedHeaders) {
+          sharedHeaders
+        }
+      }
+    }
+  }
+`);
+const UPDATE_README = gql(`
+  mutation UpdateVariantReadme($readme: String!, $name: String!, $graphId: ID!) {
+    graph(id: $graphId) {
+      variant(name: $name) {
+        updateVariantReadme(readme: $readme) {
+          id
+        }
+      }
+    }
+  }`);
+
+const UPDATE_LINTER_CONFIG = gql(`
+  mutation UpdateLinterConfiguration($changes: GraphLinterConfigurationChangesInput!, $graphId: ID!) {
+    graph(id: $graphId) {
+      updateLinterConfiguration(changes: $changes) {
+        rules {
+          rule
+        }
+      }
+    }
+  }`);
+
+function createGraph(id: string) {
+  const uniqueId = v4().replace('-', '').substring(0, 6);
```

Comment:
`.replace` will only replace the first instance of `-` in the ID, though I see you're running a `substring` here. Not sure if it matters, but in case you need to remove all `-`, use [`replaceAll`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll)

Reply:
Ya good call, I just knew 4+4 = 8 which is less than 6 and did the lazy route